src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -330,6 +330,9 @@ enum IntrinsicId
             Expression_New,
             Enum_GetValues,
             Marshal_SizeOf,
+            Marshal_PtrToStructure,
```

Comment:
`StructureToPtr<T>` as well?

Reply:
The generic versions can be analyzed without dataflow and we do that: https://github.com/dotnet/corert/blob/24f1edc938af9954eac9a3385958787ca64956d5/src/ILCompiler.Compiler/src/Compiler/UsageBasedInteropStubManager.cs#L88-L121

There's no nongeneric variant that takes a Type, so there's no way to make this with with the current dataflow analysis (we don't track object types).

Reply:
Ah ok.